# Copyright 2026 The OpenXLA Authors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================
#
# This file is automatically generated by the `generate_patch` tool.
# Do not edit directly.
diff --ruN a/stablehlo/docs/generated/chlo.md b/stablehlo/docs/generated/chlo.md
--- stablehlo/docs/generated/chlo.md
+++ stablehlo/docs/generated/chlo.md
@@ -459,8 +459,7 @@
 
 Compares `lhs` and `rhs` elementwise according to `comparison_direction`
 and `compare_type`. If unspecified, `compare_type` is FLOAT for float element
-types, SIGNED for signed element types and UNSIGNED for unsigned element
-types.
+types and NOTYPE for integer and boolean element types.
 
 See
 https://www.tensorflow.org/xla/operation_semantics#element-wise_comparison_operations.
diff --ruN a/stablehlo/docs/spec.md b/stablehlo/docs/spec.md
--- stablehlo/docs/spec.md
+++ stablehlo/docs/spec.md
@@ -6425,8 +6425,8 @@
 ```python
 def sign(x):
   if is_integer(x):
-    if compare(x, 0, LT, SIGNED): return -1
-    if compare(x, 0, EQ, SIGNED): return 0
+    if compare(x, 0, LT): return -1
+    if compare(x, 0, EQ): return 0
     return 1
   elif is_float(x):
     if is_nan(x): return NaN
diff --ruN a/stablehlo/stablehlo/conversions/linalg/tests/reduce.mlir b/stablehlo/stablehlo/conversions/linalg/tests/reduce.mlir
--- stablehlo/stablehlo/conversions/linalg/tests/reduce.mlir
+++ stablehlo/stablehlo/conversions/linalg/tests/reduce.mlir
@@ -362,7 +362,7 @@
 func.func @reduce_zero_ext(%arg0: tensor<0xi1>) -> tensor<i32> {
   %0 = stablehlo.constant dense<false> : tensor<i1>
   %1 = stablehlo.constant dense<false> : tensor<0xi1>
-  %2 = stablehlo.compare  NE, %arg0, %1, UNSIGNED : (tensor<0xi1>, tensor<0xi1>) -> tensor<0xi1>
+  %2 = stablehlo.compare  NE, %arg0, %1 : (tensor<0xi1>, tensor<0xi1>) -> tensor<0xi1>
   %3 = stablehlo.convert %2 : (tensor<0xi1>) -> tensor<0xi32>
   %4 = stablehlo.constant dense<0> : tensor<i32>
   %5 = stablehlo.reduce(%3 init: %4) across dimensions = [0] : (tensor<0xi32>, tensor<i32>) -> tensor<i32>
diff --ruN a/stablehlo/stablehlo/dialect/ChloEnums.td b/stablehlo/stablehlo/dialect/ChloEnums.td
--- stablehlo/stablehlo/dialect/ChloEnums.td
+++ stablehlo/stablehlo/dialect/ChloEnums.td
@@ -54,8 +54,8 @@
 def CHLO_COMPARISON_TYPE_NOTYPE : I32EnumAttrCase<"NOTYPE", 0>;
 def CHLO_COMPARISON_TYPE_FLOAT : I32EnumAttrCase<"FLOAT", 1>;
 def CHLO_COMPARISON_TYPE_FLOAT_TOTAL_ORDER : I32EnumAttrCase<"TOTALORDER", 2>;
-def CHLO_COMPARISON_TYPE_SIGNED : I32EnumAttrCase<"SIGNED", 3>;
-def CHLO_COMPARISON_TYPE_UNSIGNED : I32EnumAttrCase<"UNSIGNED", 4>;
+def CHLO_COMPARISON_TYPE_SIGNED : I32EnumAttrCase<"SIGNED", 3>;  // Deprecated
+def CHLO_COMPARISON_TYPE_UNSIGNED : I32EnumAttrCase<"UNSIGNED", 4>;  // Deprecated
 def CHLO_COMPARISON_TYPE_WEAK_ORDER : I32EnumAttrCase<"WEAKORDER", 5>;
 
 def CHLO_ComparisonType : I32EnumAttr<"ComparisonType",
diff --ruN a/stablehlo/stablehlo/dialect/ChloOps.td b/stablehlo/stablehlo/dialect/ChloOps.td
--- stablehlo/stablehlo/dialect/ChloOps.td
+++ stablehlo/stablehlo/dialect/ChloOps.td
@@ -776,8 +776,7 @@
   string description = [{
     Compares `lhs` and `rhs` elementwise according to `comparison_direction`
     and `compare_type`. If unspecified, `compare_type` is FLOAT for float element
-    types, SIGNED for signed element types and UNSIGNED for unsigned element
-    types.
+    types and NOTYPE for integer and boolean element types.
 
     See
     https://www.tensorflow.org/xla/operation_semantics#element-wise_comparison_operations.
diff --ruN a/stablehlo/stablehlo/dialect/StablehloEnums.td b/stablehlo/stablehlo/dialect/StablehloEnums.td
--- stablehlo/stablehlo/dialect/StablehloEnums.td
+++ stablehlo/stablehlo/dialect/StablehloEnums.td
@@ -158,8 +158,8 @@
 def STABLEHLO_COMPARISON_TYPE_NOTYPE : I32EnumAttrCase<"NOTYPE", 0>;
 def STABLEHLO_COMPARISON_TYPE_FLOAT : I32EnumAttrCase<"FLOAT", 1>;
 def STABLEHLO_COMPARISON_TYPE_FLOAT_TOTAL_ORDER : I32EnumAttrCase<"TOTALORDER", 2>;
-def STABLEHLO_COMPARISON_TYPE_SIGNED : I32EnumAttrCase<"SIGNED", 3>;
-def STABLEHLO_COMPARISON_TYPE_UNSIGNED : I32EnumAttrCase<"UNSIGNED", 4>;
+def STABLEHLO_COMPARISON_TYPE_SIGNED : I32EnumAttrCase<"SIGNED", 3>;  // Deprecated
+def STABLEHLO_COMPARISON_TYPE_UNSIGNED : I32EnumAttrCase<"UNSIGNED", 4>;  // Deprecated
 def STABLEHLO_COMPARISON_TYPE_WEAK_ORDER : I32EnumAttrCase<"WEAKORDER", 5>;
 
 def StableHLO_ComparisonType : I32EnumAttr<"ComparisonType",
diff --ruN a/stablehlo/stablehlo/tests/interpret/compare.mlir b/stablehlo/stablehlo/tests/interpret/compare.mlir
--- stablehlo/stablehlo/tests/interpret/compare.mlir
+++ stablehlo/stablehlo/tests/interpret/compare.mlir
@@ -1,9 +1,9 @@
 // RUN: stablehlo-translate --interpret -split-input-file %s
 
-func.func @compare_op_test_si64_default() {
+func.func @compare_op_test_si64_signed() {
   %lhs = stablehlo.constant dense<-2> : tensor<i64>
   %rhs = stablehlo.constant dense<-2> : tensor<i64>
-  %result = stablehlo.compare EQ, %lhs, %rhs : (tensor<i64>, tensor<i64>) -> tensor<i1>
+  %result = stablehlo.compare EQ, %lhs, %rhs, SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
   check.expect_eq_const %result, dense<true> : tensor<i1>
   func.return
 }
@@ -13,7 +13,7 @@
 func.func @compare_op_test_si64() {
   %lhs = stablehlo.constant dense<[-2, -1, 0, 2, 2]> : tensor<5xi64>
   %rhs = stablehlo.constant dense<[-2, -2, 0, 1, 2]> : tensor<5xi64>
-  %result = stablehlo.compare EQ, %lhs, %rhs, SIGNED : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
+  %result = stablehlo.compare EQ, %lhs, %rhs : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
   check.expect_eq_const %result, dense<[true, false, true, false, true]> : tensor<5xi1>
   func.return
 }
@@ -23,7 +23,7 @@
 func.func @compare_op_test_si64() {
   %lhs = stablehlo.constant dense<[-2, -1, 0, 2, 2]> : tensor<5xi64>
   %rhs = stablehlo.constant dense<[-2, -2, 0, 1, 2]> : tensor<5xi64>
-  %result = stablehlo.compare NE, %lhs, %rhs, SIGNED : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
+  %result = stablehlo.compare NE, %lhs, %rhs : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
   check.expect_eq_const %result, dense<[false, true, false, true, false]> : tensor<5xi1>
   func.return
 }
@@ -33,7 +33,7 @@
 func.func @compare_op_test_si64() {
   %lhs = stablehlo.constant dense<[-2, -1, 0, 2, 2]> : tensor<5xi64>
   %rhs = stablehlo.constant dense<[-2, -2, 0, 1, 2]> : tensor<5xi64>
-  %result = stablehlo.compare GE, %lhs, %rhs, SIGNED : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
+  %result = stablehlo.compare GE, %lhs, %rhs : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
   check.expect_eq_const %result, dense<[true, true, true, true, true]> : tensor<5xi1>
   func.return
 }
@@ -43,7 +43,7 @@
 func.func @compare_op_test_si64() {
   %lhs = stablehlo.constant dense<[-2, -1, 0, 2, 2]> : tensor<5xi64>
   %rhs = stablehlo.constant dense<[-2, -2, 0, 1, 2]> : tensor<5xi64>
-  %result = stablehlo.compare GT, %lhs, %rhs, SIGNED : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
+  %result = stablehlo.compare GT, %lhs, %rhs : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
   check.expect_eq_const %result, dense<[false, true, false, true, false]> : tensor<5xi1>
   func.return
 }
@@ -53,7 +53,7 @@
 func.func @compare_op_test_si64() {
   %lhs = stablehlo.constant dense<[-2, -1, 0, 2, 2]> : tensor<5xi64>
   %rhs = stablehlo.constant dense<[-2, -2, 0, 1, 2]> : tensor<5xi64>
-  %result = stablehlo.compare LE, %lhs, %rhs, SIGNED : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
+  %result = stablehlo.compare LE, %lhs, %rhs : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
   check.expect_eq_const %result, dense<[true, false, true, false, true]> : tensor<5xi1>
   func.return
 }
@@ -63,17 +63,17 @@
 func.func @compare_op_test_si64() {
   %lhs = stablehlo.constant dense<[-2, -1, 0, 2, 2]> : tensor<5xi64>
   %rhs = stablehlo.constant dense<[-2, -2, 0, 1, 2]> : tensor<5xi64>
-  %result = stablehlo.compare LT, %lhs, %rhs, SIGNED : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
+  %result = stablehlo.compare LT, %lhs, %rhs : (tensor<5xi64>, tensor<5xi64>) -> tensor<5xi1>
   check.expect_eq_const %result, dense<[false, false, false, false, false]> : tensor<5xi1>
   func.return
 }
 
 // -----
 
-func.func @compare_op_test_ui64_default() {
+func.func @compare_op_test_ui64_unsigned() {
   %lhs = stablehlo.constant dense<0> : tensor<ui64>
   %rhs = stablehlo.constant dense<0> : tensor<ui64>
-  %result = stablehlo.compare EQ, %lhs, %rhs : (tensor<ui64>, tensor<ui64>) -> tensor<i1>
+  %result = stablehlo.compare EQ, %lhs, %rhs, UNSIGNED : (tensor<ui64>, tensor<ui64>) -> tensor<i1>
   check.expect_eq_const %result, dense<true> : tensor<i1>
   func.return
 }
@@ -83,7 +83,7 @@
 func.func @compare_op_test_ui64() {
   %lhs = stablehlo.constant dense<[0, 1]> : tensor<2xui64>
   %rhs = stablehlo.constant dense<[0, 0]> : tensor<2xui64>
-  %result = stablehlo.compare EQ, %lhs, %rhs, UNSIGNED : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
+  %result = stablehlo.compare EQ, %lhs, %rhs : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
   check.expect_eq_const %result, dense<[true, false]> : tensor<2xi1>
   func.return
 }
@@ -93,7 +93,7 @@
 func.func @compare_op_test_ui64() {
   %lhs = stablehlo.constant dense<[0, 1]> : tensor<2xui64>
   %rhs = stablehlo.constant dense<[0, 0]> : tensor<2xui64>
-  %result = stablehlo.compare NE, %lhs, %rhs, UNSIGNED : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
+  %result = stablehlo.compare NE, %lhs, %rhs : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
   check.expect_eq_const %result, dense<[false, true]> : tensor<2xi1>
   func.return
 }
@@ -103,7 +103,7 @@
 func.func @compare_op_test_ui64() {
   %lhs = stablehlo.constant dense<[0, 1]> : tensor<2xui64>
   %rhs = stablehlo.constant dense<[0, 0]> : tensor<2xui64>
-  %result = stablehlo.compare GE, %lhs, %rhs, UNSIGNED : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
+  %result = stablehlo.compare GE, %lhs, %rhs : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
   check.expect_eq_const %result, dense<[true, true]> : tensor<2xi1>
   func.return
 }
@@ -113,7 +113,7 @@
 func.func @compare_op_test_ui64() {
   %lhs = stablehlo.constant dense<[0, 1]> : tensor<2xui64>
   %rhs = stablehlo.constant dense<[0, 0]> : tensor<2xui64>
-  %result = stablehlo.compare GT, %lhs, %rhs, UNSIGNED : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
+  %result = stablehlo.compare GT, %lhs, %rhs : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
   check.expect_eq_const %result, dense<[false, true]> : tensor<2xi1>
   func.return
 }
@@ -123,7 +123,7 @@
 func.func @compare_op_test_ui64() {
   %lhs = stablehlo.constant dense<[0, 1]> : tensor<2xui64>
   %rhs = stablehlo.constant dense<[0, 0]> : tensor<2xui64>
-  %result = stablehlo.compare LE, %lhs, %rhs, UNSIGNED : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
+  %result = stablehlo.compare LE, %lhs, %rhs : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
   check.expect_eq_const %result, dense<[true, false]> : tensor<2xi1>
   func.return
 }
@@ -133,7 +133,7 @@
 func.func @compare_op_test_ui64() {
   %lhs = stablehlo.constant dense<[0, 1]> : tensor<2xui64>
   %rhs = stablehlo.constant dense<[0, 0]> : tensor<2xui64>
-  %result = stablehlo.compare LT, %lhs, %rhs, UNSIGNED : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
+  %result = stablehlo.compare LT, %lhs, %rhs : (tensor<2xui64>, tensor<2xui64>) -> tensor<2xi1>
   check.expect_eq_const %result, dense<[false, false]> : tensor<2xi1>
   func.return
 }
diff --ruN a/stablehlo/stablehlo/tests/interpret/sort.mlir b/stablehlo/stablehlo/tests/interpret/sort.mlir
--- stablehlo/stablehlo/tests/interpret/sort.mlir
+++ stablehlo/stablehlo/tests/interpret/sort.mlir
@@ -22,7 +22,7 @@
   %input = stablehlo.constant dense<[[0, 1, 1, -3, -3, -2, 0], [4, -1, 3, 0, 4, 0, -3], [1, 0, 0, 0, 0, 2, 3], [-3, -4, -4, -2, 0, 3, 3], [0, 5, 2, -2, 0, -2, 0]]> : tensor<5x7xi64>
   %result = "stablehlo.sort"(%input) <{dimension = 0 : i64}> ({
   ^bb0(%arg0: tensor<i64>, %arg1: tensor<i64>):
-    %1 = stablehlo.compare  LT, %arg0, %arg1,  SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
+    %1 = stablehlo.compare  LT, %arg0, %arg1 : (tensor<i64>, tensor<i64>) -> tensor<i1>
     stablehlo.return %1 : tensor<i1>
   }) : (tensor<5x7xi64>) -> tensor<5x7xi64>
   check.expect_eq_const %result, dense<[[ -3, -4, -4, -3, -3, -2, -3 ], [ 0, -1, 0, -2, 0, -2, 0 ], [ 0, 0, 1, -2, 0, 0, 0 ], [ 1, 1, 2, 0, 0, 2, 3 ], [ 4, 5, 3, 0, 4, 3, 3 ]]> :  tensor<5x7xi64>
diff --ruN a/stablehlo/stablehlo/tests/transforms/config_assume_no_undeclared_side_effects.mlir b/stablehlo/stablehlo/tests/transforms/config_assume_no_undeclared_side_effects.mlir
--- stablehlo/stablehlo/tests/transforms/config_assume_no_undeclared_side_effects.mlir
+++ stablehlo/stablehlo/tests/transforms/config_assume_no_undeclared_side_effects.mlir
@@ -11,7 +11,7 @@
   // CHECK-NOT: stablehlo.while
   %1 = stablehlo.while(%iterArg = %0) : tensor<i64>
     cond {
-    %2 = stablehlo.compare LT, %0, %0, SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
+    %2 = stablehlo.compare LT, %0, %0 : (tensor<i64>, tensor<i64>) -> tensor<i1>
     stablehlo.return %2 : tensor<i1>
   } do {
     %2 = func.call @foo() : () -> tensor<i64>
diff --ruN a/stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_folder.mlir b/stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_folder.mlir
--- stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_folder.mlir
+++ stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_folder.mlir
@@ -1519,7 +1519,7 @@
   // CHECK-NOT: stablehlo.while
   %1 = stablehlo.while(%iterArg = %0) : tensor<i64>
     cond {
-    %2 = stablehlo.compare LT, %0, %0, SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
+    %2 = stablehlo.compare LT, %0, %0 : (tensor<i64>, tensor<i64>) -> tensor<i1>
     stablehlo.return %2 : tensor<i1>
   } do {
     %2 = stablehlo.custom_call @something(%iterArg) {has_side_effect = false} : (tensor<i64>) -> tensor<i64>
diff --ruN a/stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_simplification.mlir b/stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_simplification.mlir
--- stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_simplification.mlir
+++ stablehlo/stablehlo/tests/transforms/stablehlo_aggressive_simplification.mlir
@@ -224,10 +224,10 @@
   // CHECK-DAG:  [[C1:%.+]] = stablehlo.constant dense<true> : tensor<i1>
   // CHECK-DAG:  [[C5:%.+]] = stablehlo.constant dense<5> : tensor<i32>
 
-  // CHECK-DAG:  [[R0:%.+]] = stablehlo.compare EQ, [[ARG0]], [[C5]], SIGNED
-  // CHECK-DAG:  [[R1:%.+]] = stablehlo.compare GT, [[ARG0]], [[C5]], SIGNED
-  // CHECK-DAG:  [[R2:%.+]] = stablehlo.compare LE, [[ARG0]], [[C5]], SIGNED
-  // CHECK-DAG:  [[R3:%.+]] = stablehlo.compare NE, [[ARG0]], [[C5]], SIGNED
+  // CHECK-DAG:  [[R0:%.+]] = stablehlo.compare EQ, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R1:%.+]] = stablehlo.compare GT, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R2:%.+]] = stablehlo.compare LE, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R3:%.+]] = stablehlo.compare NE, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
 
   // CHECK-NEXT: return [[C1]], [[C0]], [[C1]], [[C0]], [[R0]], [[R1]], [[R2]], [[R3]]
   return %0, %1, %2, %3, %4, %5, %6, %7 :
@@ -256,24 +256,52 @@
   // CHECK-DAG:  [[C1:%.+]] = stablehlo.constant dense<true> : tensor<i1>
   // CHECK-DAG:  [[C5:%.+]] = stablehlo.constant dense<5> : tensor<i32>
 
-  // CHECK-DAG:  [[R0:%.+]] = stablehlo.compare EQ, [[ARG0]], [[C5]], UNSIGNED
-  // CHECK-DAG:  [[R1:%.+]] = stablehlo.compare GT, [[ARG0]], [[C5]], UNSIGNED
-  // CHECK-DAG:  [[R2:%.+]] = stablehlo.compare LE, [[ARG0]], [[C5]], UNSIGNED
-  // CHECK-DAG:  [[R3:%.+]] = stablehlo.compare NE, [[ARG0]], [[C5]], UNSIGNED
+  // CHECK-DAG:  [[R0:%.+]] = stablehlo.compare EQ, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R1:%.+]] = stablehlo.compare GT, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R2:%.+]] = stablehlo.compare LE, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R3:%.+]] = stablehlo.compare NE, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
 
   // CHECK-NEXT: return [[C1]], [[C0]], [[C1]], [[C0]], [[R0]], [[R1]], [[R2]], [[R3]]
   return %0, %1, %2, %3, %4, %5, %6, %7 :
          tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>
 }
 
+// CHECK-LABEL: func.func @compare_int_no_compare_type
+// CHECK-SAME:   ([[ARG0:%.+]]: tensor<i32>, [[ARG1:%.+]]: tensor<ui32>)
+func.func @compare_int_no_compare_type(%arg0: tensor<i32>, %arg1: tensor<ui32>)
+  -> (tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>) {
+  %c5 = stablehlo.constant dense<5> : tensor<i32>
+  %cu5 = stablehlo.constant dense<5> : tensor<ui32>
+
+  %0 = stablehlo.compare EQ, %arg0, %arg0 : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  %1 = stablehlo.compare LT, %arg1, %arg1 : (tensor<ui32>, tensor<ui32>) -> tensor<i1>
+  %2 = stablehlo.compare LT, %c5, %arg0 : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  %3 = stablehlo.compare GE, %cu5, %arg1 : (tensor<ui32>, tensor<ui32>) -> tensor<i1>
+  %4 = stablehlo.compare LT, %arg0, %c5, SIGNED : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  %5 = stablehlo.compare LT, %arg1, %cu5, UNSIGNED : (tensor<ui32>, tensor<ui32>) -> tensor<i1>
+
+  // CHECK-DAG:  [[C0:%.+]] = stablehlo.constant dense<false> : tensor<i1>
+  // CHECK-DAG:  [[C1:%.+]] = stablehlo.constant dense<true> : tensor<i1>
+  // CHECK-DAG:  [[C5:%.+]] = stablehlo.constant dense<5> : tensor<i32>
+  // CHECK-DAG:  [[CU5:%.+]] = stablehlo.constant dense<5> : tensor<ui32>
+  // CHECK-DAG:  [[R0:%.+]] = stablehlo.compare GT, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R1:%.+]] = stablehlo.compare LE, [[ARG1]], [[CU5]] : (tensor<ui32>, tensor<ui32>) -> tensor<i1>
+  // CHECK-DAG:  [[R2:%.+]] = stablehlo.compare LT, [[ARG0]], [[C5]] : (tensor<i32>, tensor<i32>) -> tensor<i1>
+  // CHECK-DAG:  [[R3:%.+]] = stablehlo.compare LT, [[ARG1]], [[CU5]] : (tensor<ui32>, tensor<ui32>) -> tensor<i1>
+
+  // CHECK-NEXT: return [[C1]], [[C0]], [[R0]], [[R1]], [[R2]], [[R3]]
+  return %0, %1, %2, %3, %4, %5 :
+         tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>, tensor<i1>
+}
+
 // CHECK-LABEL: func.func @compare_op_bool_simplify
 // CHECK-SAME:   ([[ARG0:%.+]]: tensor<i1>)
 func.func @compare_op_bool_simplify(%arg0: tensor<i1>) -> (tensor<i1>, tensor<i1>) {
   %false = stablehlo.constant dense<false> : tensor<i1>
   %true = stablehlo.constant dense<true> : tensor<i1>
   // CHECK-NOT: stablehlo.compare
-  %0 = stablehlo.compare NE, %arg0, %false, UNSIGNED : (tensor<i1>, tensor<i1>) -> tensor<i1>
-  %1 = stablehlo.compare EQ, %arg0, %true, UNSIGNED : (tensor<i1>, tensor<i1>) -> tensor<i1>
+  %0 = stablehlo.compare NE, %arg0, %false : (tensor<i1>, tensor<i1>) -> tensor<i1>
+  %1 = stablehlo.compare EQ, %arg0, %true : (tensor<i1>, tensor<i1>) -> tensor<i1>
   // CHECK: return [[ARG0]], [[ARG0]]
   func.return %0, %1 : tensor<i1>, tensor<i1>
 }
diff --ruN a/stablehlo/stablehlo/tests/transforms/stablehlo_target_independent_optimization.mlir b/stablehlo/stablehlo/tests/transforms/stablehlo_target_independent_optimization.mlir
--- stablehlo/stablehlo/tests/transforms/stablehlo_target_independent_optimization.mlir
+++ stablehlo/stablehlo/tests/transforms/stablehlo_target_independent_optimization.mlir
@@ -48,7 +48,7 @@
   // CHECK: stablehlo.while
   %0 = stablehlo.while(%iterArg = %arg0) : tensor<i64>
     cond {
-    %1 = stablehlo.compare LT, %iterArg, %iterArg, SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
+    %1 = stablehlo.compare LT, %iterArg, %iterArg : (tensor<i64>, tensor<i64>) -> tensor<i1>
     stablehlo.return %1 : tensor<i1>
   } do {
     %1 = stablehlo.create_token : !stablehlo.token
@@ -63,7 +63,7 @@
   // CHECK-NOT: stablehlo.while
   %0 = stablehlo.while(%iterArg = %arg0) : tensor<i64>
     cond {
-    %1 = stablehlo.compare LT, %iterArg, %iterArg, SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
+    %1 = stablehlo.compare LT, %iterArg, %iterArg : (tensor<i64>, tensor<i64>) -> tensor<i1>
     stablehlo.return %1 : tensor<i1>
   } do {
     %1 = stablehlo.create_token : !stablehlo.token
@@ -77,7 +77,7 @@
   // CHECK-NOT: stablehlo.while
   %0 = stablehlo.while(%iterArg = %arg0) : tensor<i64>
     cond {
-    %1 = stablehlo.compare LT, %arg0, %arg0, SIGNED : (tensor<i64>, tensor<i64>) -> tensor<i1>
+    %1 = stablehlo.compare LT, %arg0, %arg0 : (tensor<i64>, tensor<i64>) -> tensor<i1>
     stablehlo.return %1 : tensor<i1>
   } do {
     %1 = stablehlo.custom_call @something(%iterArg) {has_side_effect = false} : (tensor<i64>) -> tensor<i64>
diff --ruN a/stablehlo/stablehlo/transforms/optimization/StablehloAggressiveSimplification.cpp b/stablehlo/stablehlo/transforms/optimization/StablehloAggressiveSimplification.cpp
--- stablehlo/stablehlo/transforms/optimization/StablehloAggressiveSimplification.cpp
+++ stablehlo/stablehlo/transforms/optimization/StablehloAggressiveSimplification.cpp
@@ -201,10 +201,7 @@
 
     // Bail out on non-integer comparison.
     // TODO: Support more comparison types.
-    std::optional<ComparisonType> compType = op.getCompareType();
-    if (!compType ||
-        !llvm::is_contained({ComparisonType::SIGNED, ComparisonType::UNSIGNED},
-                            *compType)) {
+    if (!isa<IntegerType>(op.getLhs().getType().getElementType())) {
       return failure();
     }
 
@@ -239,15 +236,28 @@
     matchPattern(lhs, m_Constant(&lhsAttr));
     matchPattern(rhs, m_Constant(&rhsAttr));
 
+    std::optional<ComparisonType> compType = op.getCompareType();
+    bool hasRedundantCompareType =
+        compType &&
+        llvm::is_contained({ComparisonType::NOTYPE, ComparisonType::SIGNED,
+                            ComparisonType::UNSIGNED},
+                           *compType);
+
     // The canonical form has the constant operand as the RHS.
     if (lhsAttr && !rhsAttr) {
       rewriter.modifyOpInPlace(op, [&op, direction, lhs, rhs] {
         op.setComparisonDirection(invertDirection(direction));
         op->setOperands(ValueRange{rhs, lhs});
+        op.removeCompareTypeAttr();
       });
       return success();
     }
 
+    if (hasRedundantCompareType) {
+      rewriter.modifyOpInPlace(op, [&op] { op.removeCompareTypeAttr(); });
+      return success();
+    }
+
     return failure();
   }
 };
